### PR TITLE
Make Race return selected element.

### LINKED ISSUE #277: Revert to allowing multiple selectors in page Element functions
Recent API changes have made it difficult to select multiple elements at once; The previous version allowed:
```
page.MustElement("body", "title", ".new-class")
```

Now, we have to do
```
page.Race().MustElement("body", func(){...}).MustElement("title", func(){...}).MustElement(".new-class", func(){...}).MustDo()
```
which is clearly lengthy and complicates the selectors.

Two options are:
1. Bring back multiple selectors in the `Element` functions.
2. Keep the `(*page) Element*()` functions as it is, and change the `(* RaceContext) Element*()` to
    - Returning the selected element from the chained sequence.
    - Removing the function parameter (as the returned element can be used to check and run corresponding code).

I think 2 is a better option. What do you think?

### PR DESCRIPTION
**Pull request by infinitepr0**
Fix #277

### CODE REVIEW

**Comment by ysmood** [comment]
I fixed the lint.

Sorry I didn't make it clear. I mean make the callback optional, not remove it completely.

For example, if you don't use the callback you have to write the same selector twice:

image

But if you use the callback, the IDE will auto-fill the function and signature for you:

Kapture 2020-11-13 at 23 58 45

**Comment by ysmood** [comment]
To make it clear, the final code may look like this:

```go
page.Race().Element(".a").MustHandle(func(el *rod.Element) {
	// ...
}).Element(".b").MustHandle(func(el *rod.Element) {
	// ...
}).MustDo()
```

File: examples_test.go [hunk 1]
```diff
@@ -267,13 +267,17 @@ func Example_race_selectors() {
 	page.MustElement("#id_password").MustInput(password).MustPress(input.Enter)
 
 	// It will keep retrying until one selector has found a match
-	page.Race().MustElement(".nav-user-icon-base", func(el *rod.Element) {
+	elm := page.Race().Element(".nav-user-icon-base").
+		Element("[data-cy=sign-in-error]").
+		MustDo()
+
+	if elm.MustMatches(".nav-user-icon-base") {
```
Comment (ysmood) [inline examples_test.go:274]: Can we use the `MustHandle` here? so we can reduce an unnecessary global variable `elm` and repeating selectors. Also, it's better to demonstrate as many methods as possible.

**Review by ysmood — CHANGES_REQUESTED** [review]
The overall code looks good, nice job! Some changes to the example code will make it perfect.

File: examples_test.go (hunk 1 above)
Comment (infinitepr0) [reply examples_test.go:274]: Yes, will have it done!

File: query.go [hunk 2]
```diff
@@ -333,55 +333,67 @@ func (p *Page) Race() *RaceContext {
...
+func (rc *RaceContext) Do() (*Element, error) {
+	var el *Element
+	err := utils.Retry(rc.page.ctx, rc.page.sleeper(), func() (stop bool, err error) {
 		for _, branch := range rc.branches {
-			el, err := branch.condition()
+			bEl, err := branch.condition()
 			if err == nil {
-				return true, branch.callback(el)
+				el = bEl
+				return true, branch.callback(bEl)
```
Comment (ysmood) [inline query.go:388]: You can use `if branch.callback == nil` to detect if the callback is set or not, so you don't have to repeatly set empty ones `callback:  func(e *Element) error { return nil }`

**Review by ysmood — CHANGES_REQUESTED** [review]

**Review by ysmood — APPROVED** [review]

**Pull request merged**